src/licensedcode/cache.py

```diff
@@ -48,45 +43,20 @@
 cached index is safe to use across multiple processes using lock files.
 """
 
+license_index_cache_dir = create_cache_dir('license_index', dev_mode=DEV_MODE)
 index_lock_file = join(license_index_cache_dir, 'lockfile')
 tree_checksum_file = join(license_index_cache_dir, 'tree_checksums')
 index_cache_file = join(license_index_cache_dir, 'index_cache')
 
 
-_ignored_from_hash = partial(
-    ignore.is_ignored,
-    ignores={'*.pyc': 'pyc files', '*~': 'temp gedit files', '*.swp': 'vi swap files'},
-    unignores={}
-)
-
-
-def tree_checksum(tree_base_dir=src_dir, _ignored=_ignored_from_hash):
-    """
-    Return a checksum computed from a file tree using the file paths,
-    size and last modified time stamps.
-    The purpose is to detect is there has been any modification to
-    source code or data files and use this as a proxy to verify the
-    cache consistency.
-
-    NOTE: this is not 100% fool proof but good enough in practice.
-    """
-    hashable = (pth + str(getmtime(pth)) + str(getsize(pth))
-                for pth in file_iter(tree_base_dir, ignored=_ignored))
-    return md5(''.join(sorted(hashable))).hexdigest()
-
-
 LICENSE_INDEX_LOCK_TIMEOUT = 60 * 3
 
 
-# If this file exists at the root, the cache is always checked for consistency
-DEV_MODE = os.path.exists(os.path.join(root_dir, 'SCANCODE_DEV_MODE'))
-
-
 def get_or_build_index_through_cache(
         check_consistency=DEV_MODE,
         return_index=True,
         # used for testing only
-        _tree_base_dir=src_dir,
+        _tree_base_dir=None,
```

Comment:
Why this change?

Reply:
I wanted to minimize the need to import global variables. It tricky to avoid circular imports and with lower-case names, it is easy to get them mixed up with local variables.